lib/smack/RustFixes.cpp

```diff
@@ -0,0 +1,99 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+// This patches Rust programs by removing certain language specific functions,
+// enabling later optimizations.
+//
+
+#include "smack/RustFixes.h"
+#include "smack/Naming.h"
+#include "llvm/Analysis/CallGraph.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/raw_ostream.h"
+#include <set>
+#include <string>
+
+namespace smack {
+
+using namespace llvm;
+
+/*
+The main function of rust programs looks like this:
+...
+%r = call i32 @std::rt::lang_start(..., @real_main, ...)
+...
+
+This patches the main function to:
+...
+%r = 0
+call void @real_main(...)
+...
+*/
+void fixEntry(Function &main) {
+  std::vector<Instruction *> instToErase;
+
+  for (inst_iterator I = inst_begin(main), E = inst_end(main); I != E; ++I) {
+    if (auto ci = dyn_cast<CallInst>(&*I)) {
+      if (Function *f = ci->getCalledFunction()) {
+        std::string name = f->hasName() ? f->getName() : "";
+        if (name.find(Naming::RUST_ENTRY) != std::string::npos) {
```

Comment:
How come `RUST_LANG_START_INTERNAL` is not mentioned in this if condition?

Reply:
This is aimed at fixing the entry point in the main function.